apps/frontend/src/mixins/RouteMixin.ts

```diff
@@ -1,15 +1,57 @@
+import {AppInfoModule, views} from '@/store/app_info';
+import {FilteredDataModule} from '@/store/data_filters';
+import {SearchModule} from '@/store/search';
 import {Component, Vue} from 'vue-property-decorator';
 
 @Component({})
 export default class RouteMixin extends Vue {
   // get the value of the current route
-  get current_route() {
-    return this.$router.currentRoute.path.split('/')[1];
+  get currentRoute() {
+    return AppInfoModule.currentView;
   }
 
   navigateWithNoErrors(route: string): void {
+    // Saves filter state of current route before navigation
+    switch (this.currentRoute) {
+      case views.Checklist: {
+        // Save checklist filter state and clear filters before navigation
+        FilteredDataModule.setChecklistFilterState(SearchModule.searchTerm);
+        break;
+      }
+      case views.Result: {
+        // Save results filter state and clear filters before navigation
+        FilteredDataModule.setResultsFilterState(SearchModule.searchTerm);
+        break;
+      }
+      case views.Profile: {
+        // Save profiles filter state and clear filters before navigation
+        FilteredDataModule.setProfilesFilterState(SearchModule.searchTerm);
+        break;
+      }
+    }
+
```

Comment:
@em-c-rod This is the code related to saving the filter state based on the current view

Reply:
Moved to issue #5511